components/slider/script.js

```diff
@@ -59,8 +73,85 @@ class Slider extends BaseComponent {
         this.onMouseMove = this.onMouseMove.bind(this);
         this.onMouseUp = this.onMouseUp.bind(this);
         this.init = this.init.bind(this);
+
+        this.state = {
+            orientation: 'vertical',
+            step: 10,
+        };
+
+        this.stateSchema = {
+            orientation: { type: ['string'] },
+            step: { type: ['number'] },
+        };
+    }
+
+    /**
+     * Custom element lifecycle method. Called when an attribute is changed.
+     * @param {string} name
+     * @param {string} oldValue
+     * @param {string|boolean} newValue
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (!this.isRendered) return;
+
+        this.updateAttributeState(name, newValue);
     }
 
+    /**
+     * Will update the state properties
+     * @param {string} name
+     * @param {string|boolean} value
+     */
+    updateAttributeState(name, value) {
+        switch (name) {
+            case 'step':
+                this.updateStepState(parseFloat(value));
```

Comment:
Not sure why you don't use `this.updateState` directly here.

Reply:
Because I am an idiot and making one-line methods that are not necessary :D